internal/backend/db/dbgorm/utils.go

```diff
@@ -0,0 +1,21 @@
+package dbgorm
+
+import (
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+func schemaToSDK[M any, O sdktypes.Object](model *M, err error, parseFunc func(model M) (O, error)) (O, error) {
```

Comment:
I think it is weird that we pass an error to this function just so it can check it and return the translated error

Reply:
This refactoring just removes duplicated code present in almost every file in gorm - which consists of checking for model/list and error, then returning invalid object or running conversion. 4 duplicated lines in 2 places in every file.
Moreover in some places we were checking only for error and in some we were checking objects as well - so I think in terms of uniform handling of things it's important to remove duplicated code. Functions became more clear and easy to read.